File: algorithms/easy/split_strings_by_separator.ex
defmodule Solution do
  @spec split_words_by_separator(words :: [String.t()], separator :: char) :: [String.t()]
  def split_words_by_separator(words, separator) do
    separator = List.to_string([separator])

    words
    |> Enum.flat_map(fn word ->
      word |> String.split(separator) |> Enum.filter(fn w -> w != "" end)
    end)
  end
end

File: algorithms/easy/split_strings_by_separator.rs
impl Solution {
    pub fn split_words_by_separator(words: Vec<String>, separator: char) -> Vec<String> {
        let mut res = Vec::new();
        for w in words {
            for v in w.split(separator) {
                if v != "" {
                    res.push(v.to_string());
                }
            }
        }
        res
    }
}
